src/OrleansProviders/Streams/Common/Monitors/DefaultBlockPoolMonitor.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Orleans.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
```

Comment:
Sorry, I was unclear.  Interface abstraction can be used if one wants to replace the entire class, or specific functions via interface overrides.

        public class MyBlockPoolMonitor : DefaultBlockPoolMonitor, IBlockPoolMonitor
        {
                public MyBlockPoolMonitor (Logger logger) : base(logger){}
                void IBlockPoolMonitor.TrackMemoryReleased(long releasedMemoryInByte)
                { 
                        // my TrackMemoryReleased logic
                }
        }